### PR TITLE
[*] Improved the add entry dialog and added support for splitter entry.

### PR DESCRIPTION
**Pull request by Ellen141319**
[*] Fixed some bugs.

### CODE REVIEW

File: src/main/java/billgates/view/BillGatesUtilities.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    public static final List<String> CURRENCY_CODES;
+
+    /**
+     * Gets a list of available currency codes according to ISO 3166.
+     * Notes: CAD, CNY, USD are the top three currencies.
+     *
+     * @return a list of currency codes sorted in ascending alphabetical order with CAD, CNY,
+     * and USD at the beginning.
+     */
+    public static List<String> getCurrencyCodes() {
```
Comment (Lei-Tin) [inline src/main/java/billgates/view/BillGatesUtilities.java:28]: Woah this is actually pretty cool!

Comment (Lei-Tin) [reply src/main/java/billgates/view/BillGatesUtilities.java:28]: Small issue tho: 

image

Maybe we want to only include the `CurrencyCode` if it is not in `temp`.

File: src/main/java/billgates/view/gui/AddEntryDialog.java [hunk 2]
```diff
@@ -1,112 +1,108 @@
...
+
+        this.add(new ActionLabel("Currency:"));
+        this.currencyComboBox = new ActionComboBox<>(BillGatesUtilities.CURRENCY_CODES);
+        this.currencyComboBox.setAlignmentX(LEFT_ALIGNMENT);
+        this.currencyComboBox.setFont(ActionTextField.DEFAULT_FONT);
+        this.add(this.currencyComboBox);
+
+        this.add(new ActionLabel("Description:"));
+        this.descriptionField = new ActionTextField(20);
+        this.add(this.descriptionField);
```
Comment (Lei-Tin) [inline src/main/java/billgates/view/gui/AddEntryDialog.java:85]: Do we not want to `this.___.setAlignmentX(LEFT_ALIGNMENT)`on everything else?

image


The right hand side is the example where I added `LEFT_ALIGNMENT` to everything, looks a little bit cleaner IMO?

**Review by Lei-Tin — APPROVED** [review]
Looking good! 

Small issue with the currency part though. 
And also small alignment issue.

**Review by ScottCTD — APPROVED** [review]
Future TODO:

- Add a use case to get the parent entry data to fill the inherited attributes in a splitter entry.

File: src/main/java/billgates/view/BillGatesUtilities.java (hunk 1 above)
Comment (ScottCTD) [reply src/main/java/billgates/view/BillGatesUtilities.java:28]: Fixed in #96

**Pull request merged**